### PR TITLE
Refactor PGGAN and StyleGANv1

### PR DESCRIPTION
**Pull request by plyfager**

### CODE REVIEW

File: mmgen/core/evaluation/evaluator.py [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-from mmengine import BaseDataElement
 from mmengine.evaluator import BaseMetric, Evaluator
 from mmengine.model import BaseModel
 from torch.utils.data.dataloader import DataLoader
 
 from mmgen.registry import EVALUATOR
-from mmgen.typing import ForwardOutputs, ValTestStepInputs
+from mmgen.typing import ForwardOutputs
+
+PROCESSED_DATA_BATCH = Union[dict, tuple]
```
Comment (plyfager) [inline mmgen/core/evaluation/evaluator.py:12]: Note: May add to typing.

File: mmgen/core/optimizer/optimizer_constructor.py [hunk 2]
```diff
@@ -148,3 +150,104 @@ def __call__(self, module: nn.Module) -> OptimWrapperDict:
                 optimizers[f'{key}_{idx}'] = constructor(submodule.blocks[idx])
         optimizers = OptimWrapperDict(**optimizers)
         return optimizers
+
+
+@OPTIM_WRAPPER_CONSTRUCTORS.register_module()
+class PGGANOptimWrapperConstructor:
+    """OptimizerConstructor for SinGAN models. Set optimizers for each
```
Comment (plyfager) [inline mmgen/core/optimizer/optimizer_constructor.py:157]: TODO: fix docs.

File: mmgen/core/runners/loops.py [hunk 3]
```diff
@@ -87,8 +92,9 @@ def run_iter(self, idx, data_batch: Sequence[dict],
             'before_val_iter', batch_idx=idx, data_batch=data_batch)
         # outputs should be sequence of BaseDataElement
         outputs = self.runner.model.val_step(data_batch)
+        processed_data_batch = self.data_preprocessor(data_batch)
```
Comment (plyfager) [inline mmgen/core/runners/loops.py:95]: You may get `data_preprocessor` in prepare function.

File: mmgen/models/gans/progressive_growing_unconditional_gan.py [hunk 4]
```diff
@@ -227,44 +140,157 @@ def sample_from_noise(self,
...
+
+        # interpolate between real_data and fake_data
+        interpolates = alpha * real_data + (1. - alpha) * fake_data
+        interpolates = autograd.Variable(interpolates, requires_grad=True)
+
+        disc_interpolates = self.discriminator(
+            interpolates,
+            curr_scale=self.curr_scale[0],
+            transition_weight=self._curr_transition_weight)
+        gradients = autograd.grad(
```
Comment (plyfager) [inline mmgen/models/gans/progressive_growing_unconditional_gan.py:233]: TODO: How to build loss.

File: mmgen/models/gans/stylegan1.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            real_data (Tensor): Real images, used to calculate gradient
+                penalty.
+
+        Returns:
+            Tuple[Tensor, dict]: Loss value and a dict of log variables.
+        """
+
+        losses_dict = dict()
+        losses_dict['loss_disc_fake'] = F.softplus(disc_pred_fake).mean()
+        losses_dict['loss_disc_real'] = F.softplus(disc_pred_real).mean()
```
Comment (plyfager) [inline mmgen/models/gans/stylegan1.py:112]: -disc_pred_real
